### PR TITLE
Remove support for .NET Standard 2.0 and 2.1

### LINKED ISSUE #2340: Drop support for .NET Standard
- .NET Standard is EOL
- Carries a different tree of dependencies, making dependency management more complex
- Simplifies build, test and packaging (smaller matrix)
- Gets rid of special case code for .NET Standard
- Increases the ability to merge .NET and .NET Framework classes

Wonder if any platforms will suffer from this?

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #2340

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs:346]: @DavoudEshtehari Should this now be changed to: ?

return Dns.GetHostAddressesAsync(serverName, cts.Token)
    .ConfigureAwait(false).GetAwaiter().GetResult();

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs:529]: @JRahnama Unsure how this can be improved now?

**Comment by ErikEJ** [comment]
@JRahnama All net462 builds pass, despite being labelled with **netstandard2.1** - but maybe the label is misleading?

**Comment by JRahnama** [comment]
> @JRahnama All net462 builds pass, despite being labelled with **netstandard2.1** - but maybe the label is misleading?

labels are misleading, I will try to make a test pipeline for this PR today.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs:529]: Well, I think we can remove System.Memory and check the exception maybe?

Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs:529]: Sorry, I have no idea what you are referring to...

Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs:529]: System.memory is no longer used - are you suggesting I just remove the second comment also?

**Comment by ErikEJ** [comment]
@JRahnama any progress on the pipeline Update?

**Comment by ErikEJ** [comment]
@David-Engel @JRahnama Is this still on the radar?

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs
Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs:346]: My apologies for the late response. For sure, it should be.

**Comment by DavoudEshtehari** [comment]
A new file with .NET Standard support has been added in pull request #2310.

**Comment by ErikEJ** [comment]
I will merge with latest main and remove it.

**Comment by ErikEJ** [comment]
How is it going with the pipeline update??

**Comment by DavoudEshtehari** [comment]
We're going to set up a test pipeline to verify the outcome of this change. Once it's merged, we'll replace it with the current CI.

**Comment by DavoudEshtehari** [comment]
You should be able to see this [pipeline](https://sqlclientdrivers.visualstudio.com/public/_build/results?buildId=84972&view=results), which has been setup for your branch.

**Review by DavoudEshtehari — COMMENTED** [review]
LGTM! In addition to the CI pipeline, we recently pushed a few YAML files for production pipeline into the GitHub repo, and they should be updated and verified internally. We'll prepare the necessary internal changes to ensure that these PRs are merged with the least time gap.

**Comment by ErikEJ** [comment]
@DavoudEshtehari Thanks, let me know if there is anything else I need to do

**Comment by ErikEJ** [comment]
@DavoudEshtehari I have updated to latest main, hope you can help get this in soon 😄

**Review by DavoudEshtehari — APPROVED** [review]

**Comment by DavoudEshtehari** [comment]
[Pipeline](https://sqlclientdrivers.visualstudio.com/public/_build/results?buildId=84972&view=results) without net standard to verify the result.

**Comment by ErikEJ** [comment]
@JRahnama  @David-Engel Are you able to have a look now?

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj:133]: Q: Is this needed? This appears new and we aren't adding the Microsoft.Data.SqlClient.SNI package reference when targeting netfx...

**Comment by ErikEJ** [comment]
@DavoudEshtehari Ready for another attempt with the new pipeline 😄

**Review by JRahnama — APPROVED** [review]

**Comment by JRahnama** [comment]
@ErikEJ Thank you for doing this. Your support is greatly appreciated. This PR is ready to merge when the conflict is addressed. 🥳 🎉

**Comment by ErikEJ** [comment]
@JRahnama Conflicts resolved - and thanks for the kind words

**Pull request merged**
